src/Api.js

```diff
@@ -0,0 +1,82 @@
+import camelCase from 'lodash.camelcase';
+import snakeCase from 'lodash.snakecase';
+
+import App from './App';
+
+export const modifyObjectKeys = (object, modify) => {
+  // If the passed in object is not an Object, return it.
+  if (
+    object === undefined ||
+    object === null ||
+    (typeof object !== 'object' && !Array.isArray(object))
+  ) {
+    return object;
+  }
+
+  if (Array.isArray(object)) {
+    return object.map(value => modifyObjectKeys(value, modify));
+  }
+
+  // Otherwise, process all its keys.
+  const result = {};
+  Object.entries(object).forEach(([key, value]) => {
+    result[modify(key)] = modifyObjectKeys(value, modify);
+  });
+  return result;
+};
+
+export const camelCaseObject = object => modifyObjectKeys(object, camelCase);
+
+export const snakeCaseObject = object => modifyObjectKeys(object, snakeCase);
+
+export const convertKeyNames = (object, nameMap) => {
+  const transformer = key =>
+    (nameMap[key] === undefined ? key : nameMap[key]);
+
+  return modifyObjectKeys(object, transformer);
+};
+
+export function formatKnownError(error) {
```

Comment:
Maybe getting the word `user` facing errors in the name would help make this clearer

Reply:
I wasn't 100% on the name of this - they're things from the server we know how to handle and display to the user.  I think your comment about changing it to something like `formatUserFacingError` or something like that would make this clearer.